### PR TITLE
Adds some hostname + version metric

### PR DESCRIPTION
**Pull request by Garbett1**

### CODE REVIEW

File: src/metrics/prometheus.go [hunk 1]
```diff
@@ -24,6 +26,23 @@ func Push(config *core.Configuration) {
 		return
 	}
 
+	if config.Metrics.PushHostinfo {
+		name, _ := os.Hostname()
+		gauge := prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (peterebden) [inline src/metrics/prometheus.go:31]: why a gauge rather than a counter?

File: src/metrics/prometheus.go [hunk 2]
```diff
@@ -24,6 +26,23 @@ func Push(config *core.Configuration) {
...
+			Namespace: "plz",
+			Subsystem: "metrics",
+			Name:      "hostinfo",
+			Help:      "Please host running info",
+			ConstLabels: prometheus.Labels{
+				"remote":   strconv.FormatBool(config.Remote.URL != ""),
```
Comment (peterebden) [inline src/metrics/prometheus.go:37]: prefer `config.NumRemoteExecutors() > 0`, that handles both the URL and whether we're configured to execute anything

maybe we should add an explicit function for this too, there are several cases where things ask this question

File: src/core/config.go [hunk 3]
```diff
@@ -708,6 +708,7 @@ type Configuration struct {
 	Metrics struct {
 		PrometheusGatewayURL string       `help:"The gateway URL to push prometheus updates to."`
 		Timeout              cli.Duration `help:"timeout for pushing to the gateway. Defaults to 2 seconds." `
+		PushHostinfo         bool         `help:"Whether to push host info"`
```
Comment (peterebden) [inline src/core/config.go:711]: nit: `PushHostInfo`

**Review by peterebden — COMMENTED** [review]
thanks!

File: src/metrics/prometheus.go (hunk 1 above)
Comment (Garbett1) [reply src/metrics/prometheus.go]: Re-considered this, counter probably does make more sense from the perspective of gathering and summing them.

File: src/core/config.go [hunk 4]
```diff
@@ -1080,6 +1081,13 @@ func (config *Configuration) NumRemoteExecutors() int {
 	return config.Remote.NumExecutors
 }
 
+func (config *Configuration) IsRemoteExecutution() bool {
```
Comment (peterebden) [inline src/core/config.go:1084]: Execututorisation -> Execution

**Comment by Garbett1** [comment]
I'm landing despite the lint failing as I assume we might want to update the Go plugin soon.

**Pull request merged**